lib/features/mailbox_dashboard/presentation/controller/search_controller.dart

```diff
@@ -0,0 +1,130 @@
+import 'dart:async';
+
+import 'package:get/get.dart';
+import 'package:jmap_dart_client/jmap/account_id.dart';
+import 'package:jmap_dart_client/jmap/core/filter/filter_operator.dart';
+import 'package:jmap_dart_client/jmap/core/filter/operator/logic_filter_operator.dart';
+import 'package:jmap_dart_client/jmap/core/sort/comparator.dart';
+import 'package:jmap_dart_client/jmap/core/unsigned_int.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email_comparator.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email_comparator_property.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email_filter_condition.dart';
+import 'package:jmap_dart_client/jmap/mail/mailbox/mailbox.dart';
+import 'package:model/model.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/domain/state/quick_search_email_state.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/domain/usecases/quick_search_email_interactor.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/presentation/model/email_receive_time_type.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/presentation/model/quick_search_filter.dart';
+import 'package:tmail_ui_user/features/thread/domain/constants/thread_constants.dart';
+
+mixin SearchController {
+  final QuickSearchEmailInteractor _quickSearchEmailInteractor =
+      Get.find<QuickSearchEmailInteractor>();
+
+  final accountId = Rxn<AccountId>();
+  final inMailbox = Rxn<MailboxId>();
```

Comment:
Manage all of these seem complicated. Other developer could make the mistake. Please find the way to keep in single place. Please think about this controller is a function `f` in math. it is better to minimize the number of variables the function `f` depend on. `f(x)` should be better than `f(x1, x2, x3, ....)`